pkgs/stdenv/generic/check-meta.nix

```
@@ -133,6 +133,9 @@ let
       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allow${allow_attr} = true; }
       to ~/.config/nixpkgs/config.nix.
+
+      d) If you are using `nix run`, in addition to one of the fixes above, you
+      may need to use the `--impure` flag as well.
```

```suggestion
      d) If you are using nix unstable commands, in addition to one of the fixes above, you
      may need to use the `--impure` flag as well.
```
Not sure the best way to word that, but this applies to build, run, shell and other unstable commands.

---

Is that part of nix-command or flakes experimental feature?

---

I'm not sure, one of them makes all evaluation pure by default. I think its flakes experimental feature. But usually you enable both, so I don't know.

---

Sorry for the delay, @Pacman99, and thank you for the review.

I tried to make the information more comprehensive (ade15a5), but I also don't know the answer to @SuperSandro2000's question, so am not totally sure how best to phrase it. Please take a look at my update and let me know what you think.

Thanks!

---

I consider the examples problematic, `run` and `build` exist for 2.3 as well, and `shell` also IIRC.

Can't you say "in a pre release version of nix"?

---

@nobbz, could you maybe provide some example wording? I'm not sure how to update the text to resolve your concerns.

---

"If you are using a prerelease version of nix (eg. flake enabled), you need to use the `--impure` flag in combination with solution a) or c)"